internal/validate/validate.go

```diff
@@ -0,0 +1,29 @@
+package validate
+
+import (
+	"context"
+	"errors"
+
+	platformv1alpha1 "github.com/openshift/api/platform/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	errNotUniquePackage = errors.New("spec.package.name is not unique")
+)
+
+func UniquePackage(ctx context.Context, cli client.Client, po *platformv1alpha1.PlatformOperator) error {
+	existingPlatformOperators := &platformv1alpha1.PlatformOperatorList{}
+	if err := cli.List(ctx, existingPlatformOperators); !apierrors.IsNotFound(err) {
```

Comment:
Can we get a 404 during a list query? Should we just return an error here?

Reply:
I don't think we can, you're right. I thought maybe if there were not items?

Reply:
You'd perform a length check to determine whether there were zero items:

```go
if len(existingPlatformOperators.Items) == 0 {
    ...
}
```

But I don't think we'd need to perform any length checking operations here. My guess is that we wouldn't error out if there were no PO items returned by that list query, despite the PO resource being passed as a parameter as a function. We'd just implicitly accept that we have package uniqueness in this case, and we're working with an out-of-date as no error was returned.